ros_ws/src/autonomous/src/wall_following.cpp

```diff
@@ -76,8 +76,8 @@ void WallFollowing::followSingleWall(const sensor_msgs::LaserScan::ConstPtr& lid
                                     createColor(1, 0, 0, 1), 0.03);
     float wallAngle = wall.getAngle();
     this->m_debug_geometry.drawLine(2, createPoint(PREDICTION_DISTANCE, -error * leftRightSign, 0),
-                                    createPoint(PREDICTION_DISTANCE + cos(wallAngle) * 2,
-                                                (-error + sin(wallAngle) * 2) * leftRightSign, 0),
+                                    createPoint(PREDICTION_DISTANCE + (float)cos(wallAngle) * 2,
+                                                (-error + (float)sin(wallAngle) * 2) * leftRightSign, 0),
```

Comment:
I don't understand why this is needed. Since `wallAngle` is already a float, shouldn't `sin(wallAngle)` also be a float? Is this related to the literal `2`? Does using `2f` behave differently`?

Reply:
Well when i look at this: http://www.cplusplus.com/reference/cmath/sin/
it seems like you are right for c++11. However i always get a warning that says "implicit conversion from ‘float’ to ‘double’". Maybe this is an compiler issue?

Reply:
@Dadie we need your c++ wizard skillz :D

Reply:
In this case the calls to `sin` and `cos` are correctly typed `double`, as those functions are defined only for `double` in the C-Header `math.h` which is included by default by the C++ Wrapper `cmath`.

To call the C++ versions either add the namespace `std::` or ~~enrich~~ pollute the current namespace with `using std;` (**please** don't).

e.g.
```C++
constexpr float CONST_SOMETHING = 1337;
auto should_be_float = std::sin(2.0f) * 2 + CONST_SOMETHING;
```